components/kyma-environment-broker/internal/orchestration/manager/upgrade_cluster.go

```diff
@@ -71,24 +71,24 @@ func (u *upgradeClusterFactory) NewOperation(o internal.Orchestration, r orchest
 	return op.RuntimeOperation, err
 }
 
-func (u *upgradeClusterFactory) ResumeOperations(orchestrationID string) ([]orchestration.RuntimeOperation, error) {
-	ops, _, _, err := u.operationStorage.ListUpgradeClusterOperationsByOrchestrationID(orchestrationID, dbmodel.OperationFilter{States: []string{orchestration.InProgress, orchestration.Pending}})
+func (u *upgradeClusterFactory) ResumeOperations(orchestrationID string, states []string) ([]orchestration.RuntimeOperation, error) {
+	result := []orchestration.RuntimeOperation{}
+
+	ops, _, _, err := u.operationStorage.ListUpgradeClusterOperationsByOrchestrationID(orchestrationID, dbmodel.OperationFilter{States: states})
 	if err != nil {
 		return nil, err
 	}
 
-	pending := make([]orchestration.RuntimeOperation, 0)
-	inProgress := make([]orchestration.RuntimeOperation, 0)
 	for _, op := range ops {
-		if op.State == orchestration.Pending {
-			pending = append(pending, op.RuntimeOperation)
-		}
-		if op.State == orchestration.InProgress {
-			inProgress = append(inProgress, op.RuntimeOperation)
+		for _, state := range states {
```

Comment:
This is really not good, it will mess up resuming of operations when orchestration is immediate schedule. That was the reason why the in progress operations got appended before the pending ones, so that the parallel strategy gets the operations back in correct order for execution.

You could revert this back and just extend the loop logic with `Retrying`, they can come before `Pending` but after `InProgress`

Reply:
understand.. reverted the `ResumeOperations` back and created new method to collect the retrying ops.